app/services/discovery_report.rb

```diff
@@ -1,36 +1,47 @@
 # Previously a single untested 200-line method from ./lib/pre_assembly/reporting.rb
 # Takes a Bundle, enumerates report data via #each_row
 class DiscoveryReport
-  attr_reader :bundle, :start_time
+  attr_reader :bundle, :start_time, :summary
 
   delegate :bundle_dir, :content_md_creation, :manifest, :project_style, to: :bundle
-  delegate :error_count, :object_filenames_unique?, to: :bundle
+  delegate :object_filenames_unique?, to: :bundle
 
   # @param [PreAssembly::Bundle] bundle
   def initialize(bundle)
     raise ArgumentError unless bundle.is_a?(PreAssembly::Bundle)
     @start_time = Time.now
     @bundle = bundle
+    @summary = { objects_with_error: 0, mimetypes: Hash.new(0), start_time: start_time.to_s, total_size: 0 }
   end
 
   # @return [Enumerable<Hash<Symbol => Object>>]
   # @yield [Hash<Symbol => Object>] data structure about a DigitalObject
   def each_row
     return enum_for(:each_row) unless block_given?
-    bundle.objects_to_process.each { |dobj| yield process_dobj(dobj) }
+    bundle.objects_to_process.each do |dobj|
+      row = process_dobj(dobj)
+      summary[:total_size] += row[:counts][:total_size]
+      summary[:objects_with_error] += 1 unless row[:errors].empty?
+      row[:counts][:mimetypes].each { |k, v| summary[:mimetypes][k] += v }
+      yield row
+    end
   end
 
   # @param [PreAssembly::DigitalObject]
   # @return [Hash<Symbol => Object>]
   def process_dobj(dobj)
     errors = {}
+    filename_no_extension = dobj.object_files.map(&:path).select { |path| File.extname(path).empty? }
+    errors[:filename_no_extension] = filename_no_extension unless filename_no_extension.empty?
     counts = {
       total_size: dobj.object_files.map(&:filesize).sum,
-      mimetypes: Hash.new(0)
+      mimetypes: Hash.new(0),
+      filename_no_extension: filename_no_extension.count,
     }
     dobj.object_files.each { |obj| counts[:mimetypes][obj.mimetype] += 1 } # number of files by mimetype
-    errors[:filename_no_extension] = true if dobj.object_files.any? { |obj| File.extname(obj.path).empty? }
-    counts[:empty_files] = dobj.object_files.count { |obj| obj.filesize == 0 }
+    empty_files = dobj.object_files.count { |obj| obj.filesize == 0 }
+    errors[:empty_files] = empty_files if empty_files > 0
```

Comment:
do we need an error when
- the digital object has no files?  
- files are unreadable? 
- duplicate files exist?   
All these are in the old report but aren't covered here.  Perhaps you already got clarity that we don't need to check for these?  Are these Ben questions?

Reply:
Right, there are still lines 55-57:
```ruby
    errors[:empty_object] = true if counts[:total_size] > 0
    errors[:missing_files] = true unless dobj.object_files_exist?
    errors[:dupes] = true unless object_filenames_unique?(dobj)
```
The flag `missing_files` is tied to the same check, `dobj.object_files_exist?` as tied to the previous output `'missing or non-readable files'`, so there is no coverage lost.  I think those lines cover the 3 cases you mention.  The message itself is of course excluded here (as all messages) since those are delegated to a presentation layer.

Reply:
ah - missed those lines below - thanks.